File: native/mixdog-graph/tests/fixtures/parity/python/sample.py
"""Parity fixture: declaration and import shapes the graph reports for Python."""

from __future__ import annotations

import json
import os, sys
import collections.abc as abc
from pathlib import Path
from os.path import join, dirname
from . import sibling
from .helpers import load

CONSTANT = 3


def top_level(path: Path) -> str:
    def nested(value: str) -> str:
        return value

    return nested(str(path))


async def fetch_all(urls):
    return [url for url in urls]


class Service:
    registry = {}

    def __init__(self, name: str) -> None:
        self.name = name

    def run(self, payload) -> str:
        def helper(value):
            return value

        return helper(json.dumps(payload))

    @staticmethod
    def build() -> "Service":
        return Service(os.getcwd())

    class Inner:
        def ping(self):
            return sys.version


def uses(base=join(dirname(__file__), "x"), extra=(abc, sibling, load)):
    return base, extra

File: native/mixdog-graph/tests/fixtures/parity/scala/Sample.scala
// Parity fixture: declaration and import shapes the graph reports for Scala.
package com.acme.sample

import scala.collection.mutable
import scala.concurrent.{Future, ExecutionContext}
import java.nio.file._

trait Storage {
  def read(key: String): Option[String]
}

class Store(val name: String) extends Storage {
  private val cache = mutable.Map.empty[String, String]

  def read(key: String): Option[String] = cache.get(key)

  def write(key: String, value: String): Unit = cache.update(key, value)
}

object Store {
  def build(name: String): Store = new Store(name)

  def async(name: String)(implicit ec: ExecutionContext): Future[Store] =
    Future(build(name))
}

object Paths2 {
  def here: Path = Paths.get(".")
}
